experiments/capture-build-scan-info.gradle

```diff
@@ -0,0 +1,49 @@
+import org.gradle.util.GradleVersion
+
+initscript {
+    def pluginVersion = "3.6"
+
+    repositories {
+        gradlePluginPortal()
+    }
+
+    dependencies {
+        classpath("com.gradle:gradle-enterprise-gradle-plugin:${pluginVersion}")
+    }
+}
+
+def isTopLevelBuild = gradle.getParent() == null
+
+if (isTopLevelBuild) {
+    def version = GradleVersion.current().baseVersion
+    def atLeastGradle5 = version >= GradleVersion.version("5.0")
+    def atLeastGradle6 = version >= GradleVersion.version("6.0")
+    
+    if (atLeastGradle6) {
+        settingsEvaluated {
+            it.extensions["gradleEnterprise"].with {
+               buildScan {
```

Comment:
That's a good idea.

Or...would it be better to do that in a separate init script so that this one is singularly focused on capturing build scan data?  In the 2nd init script, we could make sure all of the important flags for the trial are set (like this one).

Then again, I wasn't sure how much we wanted to automate with regards to configuring their build for the trail. There's a lot to be learned by going through the motions. Maybe I should write an init script that fails the build if the things we expect to  be set aren't set yet?

Reply:
@jthurne I shared my thoughts on that via slack. Single script seems fine to me. Not failing on missing things but adding them on the fly. and Reporting non-ideal configuration and rendering it on the cmd line at the end of the experiment.